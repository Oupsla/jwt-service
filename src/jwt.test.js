import initJWTService from './jwt';

describe('jwt service', () => {
  const log = jest.fn();
  const time = jest.fn();

  afterEach(() => {
    log.mockReset();
    time.mockReset();
  });

  test('should sign data', async () => {
    time.mockReturnValueOnce(new Date('2014-01-26T00:00:00Z').getTime());

    const jwt = await initJWTService({
      JWT: {
        secret: 'secret',
        duration: '2d',
        tolerance: '2h',
        algorithms: ['HS256'],
      },
      log,
      time,
    });
    const token = await jwt.sign({
      userId: 2,
      organisationId: 3,
    });

    expect({
      token,
      logs: log.mock.calls,
      times: time.mock.calls,
    }).toMatchSnapshot();
  });

  test('should verify data', async () => {
    time.mockReturnValueOnce(new Date('2014-01-26T00:00:00Z').getTime());

    const jwt = await initJWTService({
      JWT: {
        secret: 'secret',
        duration: '2d',
        tolerance: '2h',
        algorithms: ['HS256'],
      },
      log,
      time,
    });
    const decoded = await jwt.verify(
      'eyJhbGciOiJIUzI1NiJ9.' +
        'eyJ1c2VySWQiOjIsIm9yZ2FuaXNhdGlvbklkIjozLCJpY' +
        'XQiOjEzOTA2OTQ0MDAsImV4cCI6MTM5MDg2NzIwMCwibmJmIjoxMzkwNjk0NDAwfQ.' +
        'DdWhIErffR-N-bTSsjr2tDOyinbMtYkL24IZxOVaB_0',
    );

    expect({
      decoded,
      logs: log.mock.calls,
      times: time.mock.calls,
    }).toMatchSnapshot();
  });

  test('should fail after the validity duration', async () => {
    time.mockReturnValueOnce(new Date('2014-03-30T00:00:00Z').getTime());

    const jwt = await initJWTService({
      JWT: {
        secret: 'secret',
        duration: '2d',
        tolerance: '2h',
        algorithms: ['HS256'],
      },
      log,
      time,
    });

    try {
      await jwt.verify(
        'eyJhbGciOiJIUzI1NiJ9.' +
          'eyJ1c2VySWQiOjIsIm9yZ2FuaXNhdGlvbklkIjozLCJpY' +
          'XQiOjEzOTA2OTQ0MDAsImV4cCI6MTM5MDg2NzIwMCwibmJmIjoxMzkwNjk0NDAwfQ.' +
          'DdWhIErffR-N-bTSsjr2tDOyinbMtYkL24IZxOVaB_0',
      );
      throw new Error('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      expect(err.code).toEqual('E_JWT_EXPIRED');
      expect({
        logs: log.mock.calls,
        times: time.mock.calls,
      }).toMatchSnapshot();
    }
  });
});
