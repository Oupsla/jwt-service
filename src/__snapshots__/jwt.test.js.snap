// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jwt service should fail after the validity duration 1`] = `
Object {
  "logs": Array [
    Array [
      "info",
      "JWT service initialized!",
    ],
  ],
  "times": Array [
    Array [],
  ],
}
`;

exports[`jwt service should sign data 1`] = `
Object {
  "logs": Array [
    Array [
      "info",
      "JWT service initialized!",
    ],
  ],
  "times": Array [
    Array [],
  ],
  "token": "eyJhbGciOiJIUzI1NiJ9.eyJ1c2VySWQiOjIsIm9yZ2FuaXNhdGlvbklkIjozLCJpYXQiOjEzOTA2OTQ0MDAsImV4cCI6MTM5MDg2NzIwMCwibmJmIjoxMzkwNjk0NDAwfQ.DdWhIErffR-N-bTSsjr2tDOyinbMtYkL24IZxOVaB_0",
}
`;

exports[`jwt service should verify data 1`] = `
Object {
  "decoded": Object {
    "exp": 1390867200,
    "iat": 1390694400,
    "nbf": 1390694400,
    "organisationId": 3,
    "userId": 2,
  },
  "logs": Array [
    Array [
      "info",
      "JWT service initialized!",
    ],
  ],
  "times": Array [
    Array [],
  ],
}
`;
